clippy_lints/src/methods/parse_to_string.rs

```diff
@@ -0,0 +1,12 @@
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+use rustc_span::Span;
+
+pub fn check(cx: &LateContext<'_>, expr: &Expr<'_>, span: Span) {
```

Comment:
You should at least check that the intermediate type is `String`.

Reply:
Wouldn't it make more sense to check that 

- the receiver of `to_string` isn't `&str`, since `impl ToString for &str` + `str::parse` is not actually suspicious
- the receiver of `parse` is `&str` (delegating to `FromStr`)